controller/engine_controller.go

```diff
@@ -1002,6 +1009,24 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 	return nil
 }
 
+func (m *EngineMonitor) aquireRestoringCounter(aquire bool, engine *longhorn.Engine) {
+	if !aquire {
+		m.restoringCounter.DecreaseCount()
+		return
+	}
+
+	volume, err := m.ds.GetVolumeRO(engine.Spec.VolumeName)
+	if err != nil {
+		m.logger.Warnf("Failed to get volume %v for restoring counter", engine.Spec.VolumeName)
+		return
+	}
+	isDRVolume := volume.Status.IsStandby
+	if isDRVolume {
+		return
+	}
```

Comment:
Move this to the top, because need to bypass all counting if the volume is DR.

Reply:
Moved this to the caller so it will only be checked once.